File: eBarberShop/eBarberShop.Model/Requests/PaymentDetailInsertRequest.cs
namespace eBarberShop.Model.Requests
{
    public class PaymentDetailInsertRequest
    {
        public string TransactionId { get; set; }
        public string PaymentMethod { get; set; }
        public string PayerId { get; set; }
        public string PayerFirstName { get; set; }
        public string PayerLastName { get; set; }
        public string RecipientName { get; set; }
        public string RecipientAddress { get; set; }
        public string RecipientCity { get; set; }
        public string RecipientState { get; set; }
        public int RecipientPostalCode { get; set; }
        public string RecipientCountryCode { get; set; }
        public double Total { get; set; }
        public string Currency { get; set; }
        public double Subtotal { get; set; }
        public double ShippingDiscount { get; set; }
        public string Message { get; set; }
        public DateTime CreateTime { get; set; }
        public int NarudzbaId { get; set; }
    }
}

File: eBarberShop/UI/ebarbershop_mobile/lib/models/payment_detail/payment_detail_insert_request.dart
import 'package:json_annotation/json_annotation.dart';

part 'payment_detail_insert_request.g.dart';

@JsonSerializable()
class PaymentDetailInsertRequest {
  String transactionId;
  String paymentMethod;
  String payerId;
  String payerFirstName;
  String payerLastName;
  String recipientName;
  String recipientAddress;
  String recipientCity;
  String recipientState;
  int recipientPostalCode;
  String recipientCountryCode;
  double total;
  String currency;
  double subtotal;
  double shippingDiscount;
  String message;
  DateTime createTime;
  int narudzbaId;

  PaymentDetailInsertRequest(
      this.transactionId,
      this.paymentMethod,
      this.payerId,
      this.payerFirstName,
      this.payerLastName,
      this.recipientName,
      this.recipientAddress,
      this.recipientCity,
      this.recipientState,
      this.recipientPostalCode,
      this.recipientCountryCode,
      this.total,
      this.currency,
      this.subtotal,
      this.shippingDiscount,
      this.message,
      this.createTime,
      this.narudzbaId);

  factory PaymentDetailInsertRequest.fromJson(Map<String, dynamic> json) =>
      _$PaymentDetailInsertRequestFromJson(json);

  Map<String, dynamic> toJson() => _$PaymentDetailInsertRequestToJson(this);
}
